File: src/ProjectR.Scripting/content/scripts/afflictions/Debuffs/CertainDeath.lua
function GetName()
  return "Certain Death"
end

function GetType()
  return Debuff
end

function GetHookPoints()
  return { TurnTriggered }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint == TurnTriggered then

    local deathCounter = char:GetVar("death_counter") - 1
    char:SetVar("death_counter", deathCounter)

    if deathCounter <= 0 then
      char:TakeTrueDamage(char:GetMaxHP() * 3)
      this:RemoveFrom(char)
    end
  end
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Debuffs/CertainDeath.cs
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class CertainDeath : Affliction
    {
        public override string Name { get { return "Certain Death"; } }

        public override AfflictionType Type { get { return AfflictionType.Debuff; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.TurnTriggered, 
                };
            }
        }

        protected override void OnTurnTriggered(ICharacter character)
        {
            var deathCounter = GetVar(character, "death_counter") - 1;
            SetVar(character, "death_counter", deathCounter);

            if (deathCounter <= 0)
            {
                character.TakeTrueDamage(character.CurrentHP * 10);
                RemoveFrom(character);
            }
        }
    }
}